[v1.4][F11-B2] TuitionCor

Zhu-Jiahui opened the pull request:
@olimhc @shookshire @Zhu-Jiahui

karrui commented on README.adoc:
```diff
@@ -1,7 +1,7 @@
 = Address Book (Level 4)
```
I don't think this application is named `Address Book (Level 4)` anymore!

Zhu-Jiahui replied:
updated.

karrui commented on docs/DeveloperGuide.adoc:
```diff
@@ -806,9 +806,59 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
-|`*` |user with many persons in the address book |sort persons by name |locate a person easily
+|`* *` |user |do multi-layer searching |narrow down the options
+
+|`* *` |student |request to know the qualification of the tutor |make my decision wisely
```
As discussed in the tutorial, this does not really make much sense -- would a student / tutor use this application?

Zhu-Jiahui replied:
updated

karrui commented on docs/DeveloperGuide.adoc:
```diff
@@ -806,9 +806,59 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+
+|`* *` |student |request to know the qualification of the tutor |make my decision wisely
+
+|`* *` |tutor |request to know the grades of the students |make my decision wisely
```
As above

Zhu-Jiahui replied:
updated

karrui commented on src/main/java/seedu/address/logic/commands/HelpCommand.java:
```diff
@@ -9,6 +9,7 @@
 public class HelpCommand extends Command {
 
     public static final String COMMAND_WORD = "help";
+    public static final String COMMAND_ALIAS = "1";
```
Does this alias make much sense? It does not fit in the previous established way of aliasing commands. Might be confusing to the user.

Zhu-Jiahui replied:
updated

karrui commented on docs/DeveloperGuide.adoc:
```diff
@@ -10,7 +10,7 @@ ifdef::env-github[]
-:repoURL: https://github.com/se-edu/addressbook-level4/tree/master
+:repoURL: https://github.com/CS2103JAN2018-F11-B2/main
 
 By: `Team SE-EDU`      Since: `Jun 2016`      Licence: `MIT`
```
Update this line! Preferably `XXX forked from Team SE-EDU`

olimhc replied:
Will update in my commit

karrui commented on README.adoc:
```diff
@@ -1,7 +1,7 @@
 
-https://travis-ci.org/se-edu/addressbook-level4[image:https://travis-ci.org/se-edu/addressbook-level4.svg?branch=master[Build Status]]
+https://travis-ci.org/CS2103JAN2018-F11-B2/main[image:https://travis-ci.org/se-edu/addressbook-level4.svg?branch=master[Build Status]]
 https://ci.appveyor.com/project/damithc/addressbook-level4[image:https://ci.appveyor.com/api/projects/status/3boko2x2vr5cc3w2?svg=true[Build status]]
```
Remove/ replace these badges with your team's repo instead -- they are not badges for your updated application anymore

olimhc replied:
Will update in my commit

karrui commented on README.adoc:
```diff
@@ -1,7 +1,7 @@
 = Address Book (Level 4)
```
Remember to update your application description. This application is not an Addressbook application anymore. The description should be clear as to who your target user is and what functionalities it has (or will have)

karrui commented on docs/DeveloperGuide.adoc:
```diff
@@ -806,9 +806,59 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
 
 |`* * *` |user |find a person by name |locate details of persons without having to go through the entire list
 
+|`* * *` |user |sort contacts based on their locality |better match students to tutors
```
What does locality mean? Be more specific. How will this better match students to tutors?

Zhu-Jiahui replied:
updated

karrui commented on docs/DeveloperGuide.adoc:
```diff
@@ -806,9 +806,59 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+
+|`* * *` |user |edit the information stored |keep the information up to date
+
+|`* * *` |user |tag tutors and students with multiple values |
```
The third part of the user stories is missing. All user stories should be justified with a `so that I can...` part. This is the same for all the other stories with missing information.

Zhu-Jiahui replied:
updated

karrui commented on docs/DeveloperGuide.adoc:
```diff
@@ -806,9 +806,59 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+
+|`* * *` |user |see and compare the price that students/tutors are willing to pay/accept|match them based on pricing
+
+|`* * *` |user |have an autosave function |changes will be saved once i exit the application
```
Replace the `i` with `the user`. User stories should be about the user.

Zhu-Jiahui replied:
updated

karrui commented on docs/DeveloperGuide.adoc:
```diff
@@ -806,9 +806,59 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
 |`* *` |user |hide <<private-contact-detail,private contact details>> by default |minimize chance of someone else seeing them by accident
 
-|`*` |user with many persons in the address book |sort persons by name |locate a person easily
+|`* *` |user |do multi-layer searching |narrow down the options
```
Elaborate more about this. What does multi-layer searching mean?

Zhu-Jiahui replied:
updated

karrui commented on docs/DeveloperGuide.adoc:
```diff
@@ -806,9 +806,59 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+
+|`* *` |tutor |request to know the grades of the students |make my decision wisely
+
+|`* *` |user |easily delete all students/tutors that quit with a single command |save time manually deleting
```
Don't have to elaborate on the how (the part with `with a single command`), leave that part out so that the developer would not be constrained.

Zhu-Jiahui replied:
updated

karrui commented on docs/DeveloperGuide.adoc:
```diff
@@ -806,9 +806,59 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+
+|`* *` |user |easily delete all students/tutors that quit with a single command |save time manually deleting
+
+|`* *` |user |get a list of all tutors that corresponds to a students requirements |
```
Try to merge all similar user stories together if it makes sense to do so. If not, elaborate more on the `such that I can...` part which is missing here too.

Zhu-Jiahui replied:
updated

karrui commented on docs/DeveloperGuide.adoc:
```diff
@@ -806,9 +806,59 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+|`*` |user |keep a record of how much the students improve |keep track of how good the tutor is
 |=======================================================================
 
 _{More to be added}_
```
You can remove this if there is no more to be added. Of course, keep it if you still have any. (But remember to remove it soon, you should finalise the user stories soon.)

Zhu-Jiahui replied:
updated

karrui commented on docs/DeveloperGuide.adoc:
```diff
@@ -844,16 +894,68 @@ Use case ends.
 .  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+.  Application should be able execute any command within 3 seconds.
+.  Application should have a self-explanatory installation process.
+.  Application should be less than 5mb in size.
```
Very ambitious 😛

karrui commented on docs/UserGuide.adoc:
```diff
@@ -223,11 +237,15 @@ The `redo` command fails as there are no `undo` commands executed previously.
 
+Alias Format: `x`
+
 === Saving the data
```
Might want to update the `Command Summary` below so that the existence of aliases is made known

Zhu-Jiahui replied:
updated

karrui commented on src/test/java/seedu/address/logic/parser/AddressBookParserTest.java:
```diff
@@ -103,31 +153,69 @@ public void parseCommand_history() throws Exception {
     }
 
+    @Test
+    public void parseCommand_redoCommandWord_returnsRedoCommandAlias() throws Exception {
```
Some error in the syntax here, remember that test naming syntax is `featureUnderTest_testScenario_expectedBehavior()`, so in this case it should be `parseCommand_redoCommandAlias_returnsRedoCommand()`. Note the change in position of the word **alias**

Zhu-Jiahui replied:
updated

karrui commented on src/test/java/seedu/address/logic/parser/AddressBookParserTest.java:
```diff
@@ -103,31 +153,69 @@ public void parseCommand_history() throws Exception {
     }
 
+    @Test
+    public void parseCommand_undoCommandWord_returnsUndoCommandAlias() throws Exception {
```
Same thing here

Zhu-Jiahui replied:
updated

karrui commented on src/test/java/seedu/address/ui/testutil/GuiTestAssert.java:
```diff
@@ -34,8 +37,60 @@ public static void assertCardDisplaysPerson(Person expectedPerson, PersonCardHan
+    }
+
+    /**
+     *To be updated
```
Might want to set it as a `TODO:` so that it will be highlighted in your IDE. Make sure the basic functionalities of the program is working before you push it to master, as this is the "gold standard" of your application

Zhu-Jiahui replied:
updated

karrui commented on docs/DeveloperGuide.adoc:
```diff
@@ -806,9 +806,59 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
 
```
The target user profile is not updated. Please update it so I can assess its viability or make suggestions for improvements.

olimhc replied:
Will update in my commit

karrui reviewed (commented):
[v1.0 review]
**DO NOT CLOSE THIS PR!**
Make changes as requested/suggested throughout the weeks until v1.5 :)

Few notes:
**Important:** I do not see your updated target user profile. Please remember to update them so I can assess the validity or make suggestions.
Be reminded that you will implement 1 big feature each, and it should solve the target user’s problem.
You can also define a person and give a name (to easily refer to him/her later in the doc)

Good to see that you have created issues and v1.0 milestone.
Please remember to add v1.1 to v1.5 milestones by next tutorial -- they can be empty, but should have been created with deadlines.

More issues should be created, they can be related to implementing, documentation, or bugs. I recommend using tags to sort them out.

Please implement comments you think you should follow.

karrui commented:
@Zhu-Jiahui, seems like you changed your Github username and did not update the mapping. Please contact the head TA to remap your github id.

@madsonic @olimhc please help to tell him if he does not see this.

karrui commented:
Don't need to reply to all of them... Spamming my inbox :(

Zhu-Jiahui commented:
Noted! Sorry! 
--
Sent from myMail for Android Wednesday, 14 March 2018, 11:13AM +08:00 from Kar Rui Lau  notifications@github.com :

>Don't need to reply to all of them... Spamming my inbox :(
>—
>You are receiving this because you were mentioned.
>Reply to this email directly,  view it on GitHub , or  mute the thread .

karrui commented:
Reviewing now; but a few things to take note for next submission (i.e v1.2 onwards)

Comment here when you are ready for review and give a summary of the changes -- i.e
"Add xxx function"
"Change xxx", etc
So I can see what new changes there are :)

karrui commented on README.adoc:
```diff
@@ -15,13 +15,13 @@ ifndef::env-github[]
-** A more sophisticated GUI that includes a list  panel and an in-built Browser.
-** More test cases, including automated GUI testing.
-** Support for _Build Automation_ using Gradle and for _Continuous Integration_ using Travis CI.
+* This is a desktop TuitionCor application. It has a GUI but most of the user interactions happen using a CLI (Command Line Interface).
```
This does not tell me anything about the application. What is `TuitionCor`?

Zhu-Jiahui replied:
Will update in my commit

karrui commented on docs/DeveloperGuide.adoc:
```diff
@@ -10,7 +10,7 @@ ifdef::env-github[]
-:repoURL: https://github.com/se-edu/addressbook-level4/tree/master
+:repoURL: https://github.com/CS2103JAN2018-F11-B2/main forked from Team SE-EDU
 
 By: `Team SE-EDU`      Since: `Jun 2016`      Licence: `MIT`
```
Update this line too

olimhc replied:
Will update in my commit

karrui commented on docs/DeveloperGuide.adoc:
```diff
@@ -10,7 +10,7 @@ ifdef::env-github[]
 :note-caption: :information_source:
 endif::[]
-:repoURL: https://github.com/se-edu/addressbook-level4/tree/master
+:repoURL: https://github.com/CS2103JAN2018-F11-B2/main forked from Team SE-EDU
```
Provide link to the Team SE-EDU repo too

olimhc replied:
Will update in my commit

karrui commented on docs/DeveloperGuide.adoc:
```diff
@@ -782,13 +782,25 @@ See this https://github.com/se-edu/addressbook-level4/pull/599[PR] for the step-
 *Target user profile*:
 
-* has a need to manage a significant number of contacts
+<<<<<<< Updated upstream
```
Remove merge conflict arrows

olimhc replied:
Will update in my commit

karrui commented on docs/DeveloperGuide.adoc:
```diff
@@ -782,13 +782,25 @@ See this https://github.com/se-edu/addressbook-level4/pull/599[PR] for the step-
-*Value proposition*: manage contacts faster than a typical mouse/GUI driven app
+*Value proposition*:
+TuitionCor is targeted at tuition coordinators who have to manage a large amount of contacts.
+The daily job-scope of a tuition coordinator involves the need to manage large amount of contacts and match the students to tutors according to their credentials, needs and location.
```
Good!

karrui commented on docs/DeveloperGuide.adoc:
```diff
@@ -800,18 +812,63 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+
+|`* * *` |user |tag tutors and students with multiple fields |do multi-layer searching to better match tutors and students
+
+|`* * *` |user |categorise contacts between students and tutors |reduce the number of tagging required and make searching more convenience
```
Typo here. Should be `convenient`

olimhc replied:
Will update in my commit

karrui commented on docs/DeveloperGuide.adoc:
```diff
@@ -800,18 +812,63 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+
+|`* * *` |user |see and compare the price that students/tutors are willing to pay/accept|match them based on pricing
+
+|`* * *` |user |have an auto-save function |changes will be saved once user exit the application
```
This does not seem to fit into a user story - this seems to be out of the control of the user (since it's automatic after all) and would be more suited as a NFR

olimhc replied:
Will update in my commit

karrui commented on docs/DeveloperGuide.adoc:
```diff
@@ -844,16 +901,67 @@ Use case ends.
+*MSS*
+
+1.  User types “/sortl” in to the command line.
+2.  Addressbook will prompt user to select whether they would like to view the list for students or tutors.
```
No longer `Addressbook`, remember to change all other use cases too

olimhc replied:
Will update in my commit

karrui commented on docs/DeveloperGuide.adoc:
```diff
@@ -844,16 +901,67 @@ Use case ends.
+
+[none]
+* 3a. System detects an invalid input.
++
```
Should there also be an extension for if the user wants to escape from the command input? (Such as if the user changes his mind after typing `/sortl`

olimhc replied:
Will update in my commit

karrui commented on docs/UserGuide.adoc:
```diff
@@ -247,18 +268,80 @@ _{explain how the user can enable/disable data encryption}_
 == Command Summary
 
 * *Add* `add n/NAME p/PHONE_NUMBER e/EMAIL a/ADDRESS [t/TAG]...` +
+<<<<<<< Updated upstream
```
Remove merge conflict arrows

Zhu-Jiahui replied:
Will update in my commit

karrui commented on docs/UserGuide.adoc:
```diff
@@ -247,18 +268,80 @@ _{explain how the user can enable/disable data encryption}_
+Alias: `u`
+* *Redo* : `redo` +
+Alias: `r`
+=======
```
Remove merge conflict arrows

Zhu-Jiahui replied:
Will update in my commit

karrui commented on src/main/java/seedu/address/model/person/Level.java:
```diff
@@ -0,0 +1,57 @@
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Person's related Level in the TuitionCor.
```
What exactly is a `Level`? Might want to elaborate

karrui commented on src/main/java/seedu/address/model/person/Location.java:
```diff
@@ -0,0 +1,56 @@
+public class Location {
+
+    public static final String MESSAGE_LOCATION_CONSTRAINTS =
+            "Location should only be north, south, east, west, central but can take any value if necessary";
```
You can also change regex to force the inputs to only take `North`, `South`, etc. Google around to find out more about regex.

karrui commented on src/main/java/seedu/address/model/person/Person.java:
```diff
@@ -19,11 +19,15 @@
     private final Phone phone;
     private final Email email;
     private final Address address;
+    //private final Location location;
```
Don't leave commented out code. Just go ahead and delete unused code. Now that we are using a revision control system, we can recover any deleted code later.

karrui commented on src/main/java/seedu/address/model/person/Person.java:
```diff
@@ -51,6 +55,21 @@ public Address getAddress() {
 
+
+    /*
+    public Location getLocation() {
```
You can also leave such changes on the personal branch next time as discussed in the tutorial instead of pushing it to team master branch. Remember, every version should be a better version and not include non-working functionality

karrui commented on src/main/java/seedu/address/model/person/SearchContainsKeywordsPredicate.java:
```diff
@@ -0,0 +1,37 @@
+import seedu.address.commons.util.StringUtil;
+
+/**
+ * Tests that a {@code Person}'s {@code Name} matches any of the keywords given.
```
Don't think the header matches the method anymore, remember to change headers when you refactor / change methods

karrui commented on src/main/java/seedu/address/model/person/Subject.java:
```diff
@@ -0,0 +1,57 @@
+     * The first character of the Subject must not be a whitespace,
+     * otherwise " " (a blank string) becomes a valid input.
+     */
+    public static final String ADDRESS_VALIDATION_REGEX = "[^\\s].*";
```
Remember to change such names when copy-pasting code

karrui commented on src/main/java/seedu/address/model/person/Subject.java:
```diff
@@ -0,0 +1,57 @@
+     * Returns true if a given string is a valid person Subject.
+     */
+    public static boolean isValidSubject(String test) {
+        return test.matches(ADDRESS_VALIDATION_REGEX);
```
Change the regex name here too

karrui commented on src/test/java/guitests/guihandles/HelpWindowHandle.java:
```diff
@@ -27,6 +27,7 @@ public static boolean isWindowPresent() {
 
     /**
      * Returns the {@code URL} of the currently loaded page.
+     * @TODO: 14/3/2018 Remove this when no longer required
```
Remember to clear your TODOs if they are done (I don't know if this is applicable or what this TODO is doing)
Remember to make TODOs obvious to anyone seeing your code too -- good habit to have

olimhc replied:
Will update in my commit

karrui commented on src/test/java/seedu/address/logic/parser/AddressBookParserTest.java:
```diff
@@ -48,10 +48,24 @@ public void parseCommand_add() throws Exception {
     public void parseCommand_clear() throws Exception {
-        assertTrue(parser.parseCommand(ClearCommand.COMMAND_WORD) instanceof ClearCommand);
-        assertTrue(parser.parseCommand(ClearCommand.COMMAND_WORD + " 3") instanceof ClearCommand);
+        assertTrue(parser.parseCommand(ClearCommand.COMMAND_ALIAS) instanceof ClearCommand);
```
Duplicate code? Seems like you removed the wrong lines

karrui commented on src/test/java/seedu/address/model/person/SearchContainsKeywordsPredicateTest.java:
```diff
@@ -39,37 +42,38 @@ public void equals() {
         assertTrue(predicate.test(new PersonBuilder().withName("Alice Bob").build()));
+
+        // Keywords match phone, email and address, but does not match name
+        predicate = new SearchContainsKeywordsPredicate(Arrays.asList("12345", "alice@email.com", "Main", "Street"));
```
Can also add more test cases like only phone number, only email, etc

karrui commented on src/test/java/seedu/address/ui/testutil/GuiTestAssert.java:
```diff
@@ -34,8 +37,61 @@ public static void assertCardDisplaysPerson(Person expectedPerson, PersonCardHan
+    }
+
+    /**
+     *TODO:
```
TODO alert

olimhc replied:
Will update in my commit

karrui commented on src/test/java/systemtests/SampleDataTest.java:
```diff
@@ -46,6 +46,6 @@ private void deleteFileIfExists(String filePath) {
     public void addressBook_dataFileDoesNotExist_loadSampleData() {
         Person[] expectedList = SampleDataUtil.getSamplePersons();
-        assertListMatching(getPersonListPanel(), expectedList);
+        assertListMatching(getStudentListPanel(), expectedList);
```
Might want to add test cases relating to the other panel newly introduced too

karrui reviewed (commented):
@Zhu-Jiahui @olimhc @shookshire 
v1.1 reviewed, remember to not close your PR.

**Some admin matters.**
I would want you to mention the following in your Developer guide

For each team member what are the features (major and minor) you are proposing?
Within the scope of the project, how does it fit in (just a one or 2 line summary)
Please do this by end of day Monday 19 March.

karrui commented on docs/UserGuide.adoc:
```diff
@@ -11,13 +11,12 @@ ifdef::env-github[]
 == Introduction
 
-AddressBook Level 4 (AB4) is for those who *prefer to use a desktop app for managing contacts*. More importantly, AB4 is *optimized for those who prefer to work with a Command Line Interface* (CLI) while still having the benefits of a Graphical User Interface (GUI). If you can type fast, AB4 can get your contact management tasks done faster than traditional GUI apps. Interested? Jump to the <<Qui ...
+TuitionCor is for those who *prefer to use a desktop app for managing client information*. More importantly, TuitionCor is *optimized for those who prefer to work with a Command Line Interface* (CLI) while still having the benefits of a Graphical User Interface (GUI). If you can type fast, TuitionCor can get your client management tasks done faster than traditional GUI apps. Interested? Jump to t ...
```
Please add a Features section to this UserGuide too, detailing what each new version will bring, all the way to v1.5 (and v2.0 for the long term goals)
Basically add a timeline of expected features

karrui commented on docs/DeveloperGuide.adoc:
```diff
@@ -160,7 +160,7 @@ image::UiClassDiagram.png[width="800"]
 *API* : link:{repoURL}/src/main/java/seedu/address/ui/Ui.java[`Ui.java`]
 
-The UI consists of a `MainWindow` that is made up of parts e.g.`CommandBox`, `ResultDisplay`, `PersonListPanel`, `StatusBarFooter`, `BrowserPanel` etc. All these, including the `MainWindow`, inherit from the abstract `UiPart` class.
+The UI consists of a `MainWindow` that is made up of parts e.g.`CommandBox`, `ResultDisplay`, `PersonListPanel`, `StatusBarFooter` etc. All these, including the `MainWindow`, inherit from the abstract `UiPart` class.
```
Missing documentation for new features, please remember to add them every week.

shookshire commented:
Added Feature Contribution to DeveloperGuide

karrui commented:
@Zhu-Jiahui @shookshire @olimhc  Please note the new post about defining your major and minor enhancements - https://github.com/nus-cs2103-AY1718S2/forum/issues/28
Namely, a major enhancement should span multiple stories, with stories spanning multiple tasks, and a minor enhancement should be a single story. Make sure your issue tracker and milestones are properly set up before the tutorial.

A demo issue tracker has also been released for your perusal. You may also find it at https://github.com/MyTAOrg/DemoGradle/issues

destinnsy committed:
Merge pull request #34 from olimhc/SplitListV1.1

Replace dummy text fields with "-"

olimhc committed:
Update sort implementation

olimhc committed:
Merge branch 'master' into DeveloperGuide

# Conflicts:
#	docs/DeveloperGuide.adoc

olimhc committed:
Merge pull request #56 from olimhc/DeveloperGuide

Developer guide: Sort implementation

olimhc committed:
DeveloperGuide: Update format

olimhc committed:
DeveloperGuide: Minor update

olimhc committed:
Merge pull request #57 from olimhc/DeveloperGuide

DeveloperGuide: Updates sort implementation

olimhc committed:
Included diagrams for DeveloperGuide: Sort Implementation

olimhc committed:
Merge pull request #58 from olimhc/DeveloperGuide

Include diagrams for DeveloperGuide: Sort Implementation

olimhc committed:
Merge branch 'master' into Sort

# Conflicts:
#	src/main/java/seedu/address/logic/parser/AddressBookParser.java

olimhc committed:
Edit variables naming and updated methods

karrui commented on docs/DeveloperGuide.adoc:
```diff
@@ -358,6 +359,138 @@ image::UndoRedoActivityDiagram.png[width="650"]
+An example of the input required "sort tl" which sorts tutor's list by location.
+
+[NOTE]
+If the user input format is invalid, an exception will be thrown
```
Does the exception lead to an error message being shown to the user or??

Include code.

olimhc replied:
Will update

karrui commented on docs/DeveloperGuide.adoc:
```diff
@@ -840,20 +1018,76 @@ Use case ends.
+[none]
+** 1a1. TuitionCor will display invalid command or display sort message usage.
+** 1a2. User enter required information.
+** 1a3. Steps 3a1 and 3a2 are repeated until valid input is entered.
```
`3a1`? You might be meaning `1a1`, `1a2` instead. I don't see a `3a1` :P

olimhc replied:
Will update

karrui commented on docs/DeveloperGuide.adoc:
```diff
@@ -840,20 +1018,76 @@ Use case ends.
+Use case resumes at step 2.
+
+[none]
+* 1b. System detects no further input.
```
How does it detect no further input? I don't think the functionality is in your implementation yet :P

karrui commented on docs/DeveloperGuide.adoc:
```diff
@@ -840,20 +1018,76 @@ Use case ends.
+
+1.  User enter match followed by a person name.
+2.  TuitionCor displays information of client in the respective list (either tutor list or student list)
+3.  TuitionCor returns a list of potential matches in the respective list.
```
Wrong sequence. Remember MSS are in the Actor-System-Actor-System format.

olimhc replied:
Will update

karrui commented on docs/DeveloperGuide.adoc:
```diff
@@ -840,20 +1018,76 @@ Use case ends.
+** 3a1. TuitionCor returns an empty list.
++
+Use case ends.
+
```
Where are the other use cases for your enhancements? This is also part of documentation.  @Zhu-Jiahui @shookshire @olimhc

karrui commented on docs/UserGuide.adoc:
```diff
@@ -65,24 +66,42 @@ Format: `help`
+* `a n/Betsy Crowe t/friend e/betsycrowe@example.com a/Newgate Prison p/1234567 t/criminal`
+
+=== Adding a client: `addclient`
+
```
Remember to add when this enhancement was added into the program. a {Since v1.x} will suffice. This goes for all the other enhancements.

olimhc replied:
WIll update

karrui commented on src/test/java/seedu/address/model/AddressBookTest.java:
```diff
@@ -85,6 +87,11 @@ public void getTagList_modifyList_throwsUnsupportedOperationException() {
         }
 
+        @Override
+        public ObservableList<Client> getTutorList() {
```
I don't see tests relating to this new method.

karrui commented on src/test/java/seedu/address/model/person/CategoryTest.java:
```diff
@@ -0,0 +1,38 @@
+    }
+
+    @Test
+    public void isValidCategory() {
```
wrong test naming syntax. Remember the 3 parts.

karrui commented on src/test/java/seedu/address/model/person/CategoryTest.java:
```diff
@@ -0,0 +1,38 @@
+        // null subject
+        Assert.assertThrows(NullPointerException.class, () -> Category.isValidCategory(null));
+
+        // invalid subject
```
Remember to change the copy paste comments, I don't think this is checking for `subject` anymore

karrui commented on src/test/java/seedu/address/model/person/GradeTest.java:
```diff
@@ -0,0 +1,37 @@
+    }
+
+    @Test
+    public void isValidGrade() {
```
wrong naming syntax. remember the checkstyle. This is the same for the other tests of this nature,

karrui commented on src/test/java/seedu/address/ui/testutil/GuiTestAssert.java:
```diff
@@ -34,8 +37,69 @@ public static void assertCardDisplaysPerson(Person expectedPerson, PersonCardHan
+        case "Urgent":
+            return "red";
+
+        case "UrGEnT":
```
??? Why so many variations of the word `urgent`?

olimhc replied:
Will update

karrui commented on docs/UserGuide.adoc:
```diff
@@ -27,8 +26,8 @@ AddressBook Level 4 (AB4) is for those who *prefer to use a desktop app for mana
-.  Download the latest `addressbook.jar` link:{repoURL}/releases[here].
-.  Copy the file to the folder you want to use as the home folder for your Address Book.
+.  Download the latest `TuitionCor.jar` link:{repoURL}/releases[here].
+.  Copy the file to the folder you want to use as the home folder for your client information.
```
Missing features breakdown in your UserGuide. Please add the features you expect every version to have in the form:
E.g.:
In v 1.2:  xxxxxxxxxxxxxxxxx

In v 1.3: xxxxxxxxxxxxxxxxxx

In v 1.4: xxxxxxxxxxxxxxxxxxxx

In v 1.5: xxxxxxxxxxxxxxxxxxxxxx

Also remember to say which features are coming in v2.0 (so we have an idea of the ultimate goal for your app)
Coming in v2.0: xxxxxxxxxxx

olimhc replied:
t/urgent l/north g/p1 s/math

olimhc replied:
Will update

karrui reviewed (commented):
Your current documentation still has a lot of room for improvement (both user guide and developer guide).
The architecture diagrams are outdated. Please update them. Many enhancements also have missing use cases (you are actually supposed to write out the use cases before you implement your enhancements so you have the requirements in mind).
Please put more effort into it because your documentation will be used in the grading.
 
Some other reminders:

- Please make sure that each person has written a section on implementation in the Developer Guide for their major feature.
- Tag each feature in the User Guide with (“Since v1.x”) asap.
- Remember to build a .jar file for v1.3. We will be downloading the fresh copy of it from your releases to demo in class. It must be able to run all your enhancements.
- Use UML diagrams in the Developer Guide when explaining the implementation. (There is an IntelliJ plugin called SequenceDiagram that may help.)
- Some comments that call for changes in code are still undone.

**IMPORTANT**
Please make sure to tag your repo with the correct version (e.g. for last tutorial the commits up till v1.2 should be tagged v1.2). If you are not sure how, see the textbook (W3.9c). I will be checking this as well.

This is so I can see what commits correspond to which version.

destinnsy committed:
Allowed for tutors to be saved into storage

destinnsy committed:
Revert "Allowed for tutors to be saved into storage"

This reverts commit 1bfff25e2cff74c78d4654bf534c53b4f9bd4c88.

destinnsy committed:
Revert "Revert "Allowed for tutors to be saved into storage""

This reverts commit e567bd9e276f8761ef60853a4db584c6b2b9cad1.

destinnsy committed:
XmlSerializableAddressBook; edited to record only student and tutor list

destinnsy committed:
ReadOnlyAddressBook; edited to only handle client

destinnsy committed:
Edited TuitionCor to saven and display only student and tutor list

destinnsy committed:
Checkstyle Fixes

destinnsy committed:
Changes to Test to fit UI changes

Zhu-Jiahui committed:
Merge pull request #64 from shookshire/UIComplete

Ui complete and save only tutor and student list

destinnsy committed:
Delete command changed to client

olimhc committed:
Merge branch 'WC1.3' of https://github.com/CS2103JAN2018-F11-B2/main into Sort

# Conflicts:
#	src/main/java/seedu/address/model/ModelManager.java

destinnsy committed:
Changed edit to work on client

destinnsy committed:
Changed ListCommand to work on clients

destinnsy committed:
Changed find to work on clients

olimhc committed:
Integrate sort by name command

destinnsy committed:
Edit Match to work on clients

destinnsy committed:
Removed Person related commands

destinnsy committed:
Fix most junit test

olimhc committed:
Sort functions are all working for both tutor and student list

olimhc committed:
Merge pull request #65 from shookshire/Changing_all_commands_to_Client

Changing all commands to client

olimhc committed:
TypcialPersons: Update their grade

olimhc committed:
Update PersonListPanelHandle

olimhc committed:
Remove Catergory from PersonCardHandle and remove assertEquals for catergory @ GuiTestAssert.jave.

olimhc committed:
Add dummy switchcommand

olimhc committed:
Merge branch 'WC1.3' of https://github.com/CS2103JAN2018-F11-B2/main into CloseFunc

# Conflicts:
#	src/main/java/seedu/address/logic/parser/AddressBookParser.java

olimhc committed:
Merge pull request #71 from olimhc/JunitTest

Junit test, cleared statecheck

Zhu-Jiahui committed:
minor update on match function: if the match client is a student, student list will only show that client. this is to avoid confusion.

Zhu-Jiahui committed:
Tidy up

olimhc committed:
Close function is able to close tutors and students

Zhu-Jiahui committed:
Work in progress to rank the client lists

Zhu-Jiahui committed:
Update user stories

olimhc committed:
Corrections for V1.2

Zhu-Jiahui committed:
Update developer guide

olimhc committed:
Edit CheckStyles and replaced 'Junior College' with 'Tertiary'

Zhu-Jiahui committed:
Fix checkStyle error

olimhc committed:
Edit ListCommand, FindCommand, ModelManager

Zhu-Jiahui committed:
Update display image for User Guide

olimhc committed:
Merge branch 'WC1.3' of https://github.com/CS2103JAN2018-F11-B2/main into Sort

olimhc committed:
Merge pull request #72 from Zhu-Jiahui/Enhance-find-function

Minor update on Match function and update developer guide

olimhc committed:
Merge pull request #75 from olimhc/Corrections

Corrections for V1.2 Fixes #74

olimhc committed:
Merge branch 'WC1.3' into Sort

olimhc committed:
Merge pull request #76 from olimhc/Sort

SortCommand

olimhc committed:
Removed AddCommand, cleared XMLSerializableAddressBook

destinnsy committed:
Image changed

destinnsy committed:
Continuation for Destin's DeveloperGuide update

olimhc committed:
Merge pull request #77 from olimhc/Integration

Removed AddCommand, cleared XMLAddressBookStorageTest

olimhc committed:
Edit SampleDataUtil file and typicalPersonsAddressBook.xml

olimhc committed:
Merge pull request #78 from shookshire/DeveloperGuide

Developer guide update for Destin's section

olimhc committed:
Merge pull request #79 from olimhc/Integration

Edit SampleDataUtil file and typicalPersonsAddressBook.xml

destinnsy committed:
Fix some Junit

olimhc committed:
Checkstyle fixes

olimhc committed:
Merge branch 'WC1.3' of https://github.com/CS2103JAN2018-F11-B2/main into Integration

olimhc committed:
Merge pull request #80 from olimhc/Integration

Checkstyle fixes

destinnsy committed:
More Fixes

destinnsy committed:
Fix all Junit fails

destinnsy committed:
Checkstyle fixes

destinnsy committed:
Merge pull request #81 from shookshire/WC1.3

wc1.3 complete

olimhc committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-F11-B2/main into CloseFunc

# Conflicts:
#	src/main/java/seedu/address/logic/parser/AddressBookParser.java
#	src/main/java/seedu/address/model/Model.java
#	src/main/java/seedu/address/model/ModelManager.java

olimhc committed:
Switch function works and results will be displayed

olimhc committed:
Add restore command

olimhc committed:
Disabled commands which are not available in closed/active view

olimhc committed:
Addressbook checks for duplicate in ClosedList & ActiveList

olimhc committed:
Update commands to work with ClosedList

olimhc committed:
CheckStyle Fixes and minor Junit test update

olimhc committed:
Added Junit test for sorting commands. Fixes #87

olimhc committed:
Edit minor naming errors

Zhu-Jiahui committed:
Implement Ranking function to rank the matched clients from the most suitable to the least suitable.

olimhc committed:
Update sortcommand

Zhu-Jiahui committed:
Move phone, address and email attributes higher up on the displace since they are more often used attributes

Zhu-Jiahui committed:
implement dummy highlight feature to highlight the matched attributes

destinnsy committed:
Added RemoveCommand

olimhc committed:
CheckStyle Fixes

destinnsy committed:
Edited client related fields to automatically remove multiple spaces inbetween

olimhc committed:
Update SortCommand

olimhc committed:
UserGuide & DeveloperGuide: Updated according to the new sortcommand format

olimhc committed:
UserGuide: Slight minor format changes

olimhc committed:
Include CommmadAlias for switch, close and restore

Zhu-Jiahui committed:
Implement working highlight feature to highlight the matched attributes
Implement working ranking feature to rank the matched clients from the most matched attributes to the least matched attributes

olimhc committed:
Merge pull request #88 from olimhc/Sort

Added Junit test for sorting commands. Fixes #87

Zhu-Jiahui committed:
minor changes to the code

destinnsy committed:
Added Remove Junit test and fixed broken tests

olimhc committed:
UserGuide & DeveloperGuide: Update relevant parts for switch, restore and close command.

olimhc committed:
Merge branch 'WC1.4' of https://github.com/CS2103JAN2018-F11-B2/main into CloseFunc

# Conflicts:
#	src/main/java/seedu/address/logic/commands/SortByGradeCommand.java
#	src/main/java/seedu/address/logic/commands/SortByLocationCommand.java
#	src/main/java/seedu/address/logic/commands/SortByNameCommand.java
#	src/main/java/seedu/address/logic/commands/SortBySubjectCommand.java
#	src/main/java/seedu/address/logic/commands/SortCommand.java

destinnsy committed:
checkstyle fixes

destinnsy committed:
Updated RemoveCommand in UserGuide

destinnsy committed:
checkstyle fixes

olimhc committed:
Add Junit test for switch, restore and close.

olimhc committed:
Merge pull request #89 from shookshire/WC1.4

Wc1.4 RemoveCommand and Junit fixes

olimhc committed:
Fixes Junit test and minor UserGuide update

olimhc committed:
Merge branch 'WC1.4' of https://github.com/CS2103JAN2018-F11-B2/main into CloseFunc

# Conflicts:
#	src/main/java/seedu/address/logic/parser/AddressBookParser.java

olimhc committed:
Merge pull request #90 from olimhc/Sort

Fixes Junit test and minor UserGuide update

olimhc committed:
Merge branch 'WC1.4' of https://github.com/CS2103JAN2018-F11-B2/main into CloseFunc

olimhc committed:
Minor update to TypicalClients. Fixes #35

olimhc committed:
Fixes Checkstyle

olimhc committed:
Bugs fixes.

olimhc committed:
Checkstyle fixes and minor test case updates

Zhu-Jiahui committed:
update code to pass all test cases

Zhu-Jiahui committed:
fix travis fail

Zhu-Jiahui committed:
fix checkstyle error

Zhu-Jiahui committed:
Merge pull request #92 from Zhu-Jiahui/Ranking

Ranking feature and highlighting feature

olimhc committed:
Include status for current display and update junit

olimhc committed:
Merge branch 'WC1.4' of https://github.com/CS2103JAN2018-F11-B2/main into CloseFunc

# Conflicts:
#	src/main/java/seedu/address/model/ModelManager.java

olimhc committed:
Merge pull request #91 from olimhc/CloseFunc

Close, Restore, Switch function.

olimhc committed:
Collate code

Zhu-Jiahui committed:
Collate code

olimhc committed:
Merge pull request #94 from olimhc/Integration

Collate code for V1.4

olimhc committed:
Merge pull request #95 from Zhu-Jiahui/Ranking

Collate code

olimhc committed:
UserGuide: Update

olimhc committed:
Merge pull request #96 from olimhc/Integration

UserGuide: Update

Zhu-Jiahui committed:
Update UserGuide

olimhc committed:
Update SampleDataUtil

Zhu-Jiahui committed:
Merge branch 'WC1.4' into Ranking

olimhc committed:
Fixes Travis fail

olimhc committed:
Merge pull request #98 from Zhu-Jiahui/Ranking

Update UserGuide

olimhc committed:
Merge pull request #97 from olimhc/Integration

Update SampleDataUtil

destinnsy committed:
Code Collate

destinnsy committed:
Junit Collate

destinnsy committed:
Fixed merge conflicts

olimhc committed:
Merge pull request #99 from shookshire/Collate

Collate

olimhc committed:
UserGuide: Updates

destinnsy committed:
Update stories

olimhc committed:
Merge pull request #100 from olimhc/Integration

UserGuide: Updates

olimhc committed:
UserGuide: Minor change

olimhc committed:
Merge pull request #102 from shookshire/master

Update DeveloperGuide

olimhc committed:
Merge pull request #103 from olimhc/Integration

UserGuide: Minor change

olimhc committed:
Create collated folder

Contains collated codes for V1.4

Zhu-Jiahui committed:
Delete collated

olimhc committed:
Collate files

olimhc committed:
Merge pull request #104 from olimhc/Integration

Collate files

olimhc committed:
Merge pull request #105 from CS2103JAN2018-F11-B2/WC1.4

W1.4 completed

olimhc committed:
ReadMe: Removed Learning Outcomes

olimhc committed:
ContactUs: Update with our info

olimhc committed:
UserGuide: Spelling error

olimhc committed:
Merge pull request #106 from olimhc/Integration

Remove LO, update contact us

olimhc committed:
UserGuide: Update

olimhc committed:
Merge pull request #107 from olimhc/Integration

UserGuide: Update

olimhc committed:
UserGuide: Minor change

olimhc committed:
Merge pull request #108 from olimhc/Integration

UserGuide: Minor change

olimhc committed:
Fixes #109, #112, #113, #118, #119

olimhc committed:
Partially fixes #120

Zhu-Jiahui committed:
Now match function matches attributes case insensitively

olimhc committed:
Fixes find command not able to search mutiple grade fields

olimhc committed:
Fixes bugs in grades and change app title to "TuitionCor"

olimhc committed:
Merge pull request #121 from olimhc/Integration

Fixes issue

Zhu-Jiahui committed:
Merge branch 'WC1.5' of https://github.com/CS2103JAN2018-F11-B2/main into V1.5

olimhc committed:
ListPanelController: Add logger

Zhu-Jiahui committed:
Fixes bugs and add test cases for match related classes

Zhu-Jiahui committed:
Fix travis fail

Zhu-Jiahui committed:
Fix travis fail

olimhc committed:
Edit code comments

olimhc committed:
Fixes #114 and minor code edit

olimhc committed:
Minor codefix

Zhu-Jiahui committed:
Fix checkstyle error

olimhc committed:
Merge branch 'Integration' of https://github.com/olimhc/main into Integration

olimhc committed:
SampleData: Added more clients

olimhc committed:
Edit comments

olimhc committed:
UserGuide, DeveloperGuide: Minor changes

destinnsy committed:
WC1.5 Editing reported bugs and Updating of destin's portfolio

destinnsy committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-F11-B2/main

Zhu-Jiahui committed:
Add Jiahui Portfolio

olimhc committed:
Project Portfolio for hong cho

olimhc committed:
Fixes checkstyle errors

olimhc committed:
Merge pull request #122 from Zhu-Jiahui/V1.5

Fix bugs and update test cases

olimhc committed:
Merge pull request #123 from shookshire/master

Editing reported bugs and updating of PPP

olimhc committed:
Merge pull request #124 from olimhc/Integration

Fixes issues, includes PPP and updates for sampledata

olimhc committed:
Collated codes

olimhc committed:
Merge pull request #125 from olimhc/Integration

W1.5rc complete

olimhc committed:
Merge pull request #126 from CS2103JAN2018-F11-B2/WC1.5

Merge WC1.5 to master

olimhc committed:
Fix Ui not displaying properly

olimhc committed:
Merge pull request #127 from olimhc/Integration

Fixes Ui.PNG not displaying properly

olimhc committed:
Fixes Ui not displaying correctly

olimhc committed:
Merge pull request #128 from olimhc/Integration

Fixes Ui not displaying correctly

olimhc committed:
Fixes Ui not displaying correctly

olimhc committed:
Merge pull request #129 from olimhc/Integration

ReadMe: Fixes Ui not displaying correctly

olimhc committed:
Fixes bug of PPP not displaying properly

olimhc committed:
Remove irrelevant badges

olimhc committed:
Merge pull request #130 from olimhc/Integration

Fixes bug of PPP not displaying properly

Zhu-Jiahui committed:
Update developer guide and portfolio

Zhu-Jiahui committed:
update developer guide

olimhc committed:
Update DeveloperGuide and UserGuide

olimhc committed:
Update README, AboutUs and PPP

olimhc committed:
Merge pull request #131 from Zhu-Jiahui/V1.5

Update User guide, developer guide and portfolio

olimhc committed:
Merge branch 'WC1.5F' into Integration

olimhc committed:
Merge pull request #132 from olimhc/Integration

Updates DeveloperGuide, UserGuide, README, AboutUs & PPP

destinnsy committed:
Updates to junit and PPP

Zhu-Jiahui committed:
Update portfolio

destinnsy committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-F11-B2/main

# Conflicts:
#	docs/UserGuide.adoc

Zhu-Jiahui committed:
Merge pull request #133 from Zhu-Jiahui/V1.5

Update portfolio

destinnsy committed:
update PPP

destinnsy committed:
Merge pull request #137 from shookshire/master

Increase Coverage in Junit and updated PPP

olimhc committed:
Revision of code for Sort Commands

olimhc committed:
Added Junit test: SortCommandParserTest

olimhc committed:
Adds Junit for CloseCommandParserTest and RestoreCommandParserTest. Minor check style edits.

destinnsy committed:
update to developerguide and PPP

destinnsy committed:
Merge pull request #138 from shookshire/master

update to developerguide and PPP

destinnsy committed:
Updated PPP

destinnsy committed:
Merge pull request #139 from shookshire/master

Updated PPP

olimhc committed:
Merge branch 'master' into WC1.5F

olimhc committed:
Merge pull request #140 from CS2103JAN2018-F11-B2/WC1.5F

Merge Wc1.5F to master

olimhc committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-F11-B2/main into Integration

olimhc committed:
Adds some minor Junit test

olimhc committed:
Updates collated files

olimhc committed:
Merge pull request #141 from olimhc/Integration

Adds Junit and updates collated files

Zhu-Jiahui committed:
update sequence diagram

Zhu-Jiahui committed:
Update PPP

olimhc committed:
ReadME & DeveloperGuide: Revision

Zhu-Jiahui committed:
Merge pull request #142 from Zhu-Jiahui/V1.5

update sequence diagram

olimhc committed:
UserGuide, DeveloperGuide and PPP minor edits

olimhc committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-F11-B2/main into Integration

olimhc committed:
Merge pull request #143 from olimhc/Integration

DeveloperGuide, UserGuide and PPP Updates

olimhc committed:
Removed sorting form PPP

olimhc committed:
Merge pull request #144 from olimhc/Integration

Removed sorting form PPP

destinnsy committed:
Update DeveloperGuide

destinnsy committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-F11-B2/main

destinnsy committed:
Update userGuide

olimhc committed:
Update UI image

destinnsy committed:
Merge pull request #146 from shookshire/master

Update UGDG

olimhc committed:
Updates UserGuide & README

olimhc committed:
Merge pull request #147 from olimhc/Integration

Updates UI image

destinnsy committed:
Update remove manual testing

destinnsy committed:
minor updates to developerGuide

destinnsy committed:
fix errors in UGDG

destinnsy committed:
Merge pull request #148 from shookshire/master

Update remove manual testing

destinnsy committed:
update PPP

olimhc committed:
Merge pull request #150 from shookshire/master

update PPP

olimhc committed:
DeveloperGuide: Updates Instructions for manual testing

olimhc committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-F11-B2/main into Integration

olimhc committed:
DeveloperGuide: Added one user story

olimhc committed:
Merge pull request #151 from olimhc/Integration

DeveloperGuide: Update instruction for testing & add one user story

olimhc committed:
Fixes Ui.png not displaying properly

olimhc committed:
Merge pull request #152 from olimhc/Integration

Fixes Ui.png not displaying properly

olimhc committed:
DeveloperGuide: Minor formatting edit

olimhc committed:
Merge pull request #153 from olimhc/Integration

DeveloperGuide: Minor formatting edit

destinnsy committed:
final update to ppp

olimhc committed:
Merge pull request #154 from shookshire/master

final update to ppp

olimhc committed:
ReadMe: Fixes Rendering Error

olimhc committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-F11-B2/main into Integration

olimhc committed:
ReadMe: Minor change

olimhc committed:
Merge pull request #155 from olimhc/Integration

Fixes rendering error
